qiskit/ignis/experiments/calibration/instruction_data/gate_and_schedules_definition.py

```diff
@@ -19,31 +34,39 @@ class InstructionsDefinition:
     The class allows users to define single-pulse schedules as quantum circuits and
     then leverage the quantum circuit compose operation to compose more complex
     circuits from basic circuits.
-
-    # TODO
-    - test recursive behavior of instructions
     """
-
-    def __init__(self, backend_name: str, n_qubits: int, channel_qubit_map: Dict,
-                 pulse_table: Optional[PulseTable] = None):
+    TABLE_COLS = ['gate_name', 'qubits', 'signature', 'gate_id', 'program']
+
+    def __init__(self,
+                 backend_name: str,
+                 n_qubits: int,
+                 channel_qubit_map: Dict[str, Tuple[int]],
+                 pulse_table: Optional[PulseParameterTable] = None,
+                 parametric_shapes: Optional[Enum] = None,
+                 pulse_channels: Optional[Enum] = None):
         """
         Args:
             backend: The backend for which the instructions are defined.
         """
 
         # Dict to store the instructions we are calibrating
-        self._instructions = {}
+        self._instructions = pd.DataFrame(index=[], columns=InstructionsDefinition.TABLE_COLS)
 
         # Table to store the calibrated pulse parameters
         self.backend_name = backend_name
 
         if pulse_table:
             self._pulse_table = pulse_table
         else:
-            self._pulse_table = PulseTable()
+            self._pulse_table = PulseParameterTable(channel_qubit_map=channel_qubit_map)
 
         self._n_qubits = n_qubits
-        self._channel_map = channel_qubit_map
+
+        self._parametric_shapes = parametric_shapes or utils.ParametricPulseShapes
+        self._pulse_channels = pulse_channels or utils.ChannelPrefixes
+
+        # parameter set used to construct schedule
+        self._series = 'default'
```

Comment:
So this allows us to switch context? Does this refer to the `signature` column of `PulseParameterTable`? If so can we make it consistent and describe this better. If not than I missunderstood something.

Reply:
There is no `signature` column in the parameter table. This refers to the `series` column in the parameter table so they are consistent.

Reply:
Okay, so I misunderstood its purpose then. Are there better names we could use? `series` overlaps a bit with what we have in `Analysis`.